### PR TITLE
feat(query): added ALB Deletion Protection Disabled for Terraform

### PR DESCRIPTION
**Pull request by tiagocruz20**
**Proposed Changes**
- added a query to check if deletion_protection is enabled in Application Load Balancer (ALB)

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/aws/alb_deletion_protection_disabled/test/positive.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+resource "aws_lb" "positive" {
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/aws/alb_deletion_protection_disabled/test/positive.tf:1]: name the file positive1.tf
```suggestion
resource "aws_lb" "positive1" {
```

File: assets/queries/terraform/aws/alb_deletion_protection_disabled/test/positive_expected_result.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "queryName": "ALB Deletion Protection Disabled",
+    "severity": "LOW",
+    "line": 7,
+    "fileName": "positive.tf"
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/aws/alb_deletion_protection_disabled/test/positive_expected_result.json:6]:
```suggestion
    "fileName": "positive1.tf"
```

File: assets/queries/terraform/aws/alb_deletion_protection_disabled/metadata.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "id": "8bbb242f-6e38-4127-86d4-d8f0b2687ae2",
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/aws/alb_deletion_protection_disabled/metadata.json:2]: Create an unique query ID, see unit tests error:
```
   === RUN   TestUniqueQueryIDs
      queries_test.go:62: 
          	Error Trace:	queries_test.go:62
          	Error:      	Should be false
          	Test:       	TestUniqueQueryIDs
          	Messages:   	
          	            	non unique queryID found uuid: 8bbb242f-6e38-4127-86d4-d8f0b2687ae2
          	            	queryDir: ..\assets\queries\terraform\aws\ami_not_encrypted
          	            	duplicateDir: ..\assets\queries\terraform\aws\alb_deletion_protection_disabled
```

File: assets/queries/terraform/aws/alb_deletion_protection_disabled/metadata.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  "queryName": "ALB Deletion Protection Disabled",
+  "severity": "LOW",
+  "category": "Insecure Configurations",
+  "descriptionText": "Application Load Balancer should have deletion protection enabled",
+  "descriptionUrl": "https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb#enable_deletion_protection",
+  "platform": "Terraform",
+  "descriptionID": "248fe368",
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/aws/alb_deletion_protection_disabled/metadata.json:9]: Create a unique description ID for the new query.

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

File: assets/queries/terraform/aws/alb_deletion_protection_disabled/query.rego [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package Cx
+
+import data.generic.common as common_lib
+
+CxPolicy[result] {
+	lb := input.document[i].resource.aws_lb[name]
```
Comment (rafaela-soares) [inline assets/queries/terraform/aws/alb_deletion_protection_disabled/query.rego:6]: Maybe we can also include the resource `aws_alb`

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]

**Pull request merged**